test/test_generation.py

```diff
@@ -123,3 +123,53 @@ def test_fourier_synthesis_linescan_hrms_more_wavevectors():
     #                         - np.mean(realised_rms_heights))**2)))
     assert abs(np.mean(realised_rms_heights) -
                ref_height) / ref_height < 0.1  #
+
+
+def test_fourier_synthesis_3D():
+    n = [101, 101, 101]
+    nx, ny, nz = n
+    s = [101, 101, 101]
+    sx, sy, sz = s
+    g = [sx/nx, sy/ny, sz/nz]
+    gx, gy, gz = g
+    hurst = 0.8
+    rms_height = 0.5
+
+    x = np.arange(0, sx, gx)
+    y = np.arange(0, sy, gy)
+    z = np.arange(0, sz, gz)
+    X, Y = np.meshgrid(x, y)
+
+    c = fourier_synthesis((nx, ny, nz), (sx, sy, sz),
+                          hurst, rms_height=rms_height,
+                          long_cutoff=sx / 3.)
+
+    hursts = np.zeros(nx)
+    rmss = np.zeros(nx)
+    for i in range(nx):
+        physical_sizes = np.array([sy, sz])
+        topo = c[i, :, :]
+        rmss[i] = rms(topo)
+
+    mean_rms_x = np.mean(rmss)
+    assert(((mean_rms_x - rms_height) / rms_height) < 0.5)
```

Comment:
Ok interesting. So that would mean that the formula for the expecte RMS-height is still valid in 3D, except that it corresponds to the RMS-height of slices. 

I trust you on that.

Is the relative difference smaller on bigger surfaces ?

Reply:
It's actually not exactly the same. The rms height should simply be `np.sqrt(np.sum(c*c))`

Reply:
That is what is computed in [rms](https://github.com/ComputationalMechanics/RandomFields/blob/1820012e942b6b098791d1398742b1f380adbc77/RandomFields/Analysis/scalar_parameters.py) . 

With "expected RMS-height" I meant the formula based on the self-affine PSD to determine the self-affine prefactor.     https://github.com/ComputationalMechanics/RandomFields/blob/1820012e942b6b098791d1398742b1f380adbc77/RandomFields/Generation/fourier_synthesis.py#L212

Reply:
Okay. I'll have to look into this, not sure whether it still applies. The code at present should only work in 2D, but it should be straightforward to generalize to nD

Reply:
My approach for now is to only allow "rms_height" etc.. prescription in 2D. Otherwise use the self-affine prefactor.

Reply:
The thing that what I wanted to test here is that in all different directions (planes) we have sound generated random field as we make sections fo the domain that the 3D field belongs to

Reply:
I agree that the RMS could be also simply obtained as the RMS of the whole field

Reply:
My point is that this is *not* the rms of the whole field. It would be the rms if you took the sqrt of the mean of the squares, but you are taking the mean of the rms.

Reply:
So, I should rename the vaiables that I am checking

Reply:
I think it is more an API problem. Currently your test is showing that the parameter `rms_height` corresponds to the rms_height of 2D slices. What we want is that it will correspond to the rms-height of the 3D field.

Reply:
I added the same assertion for the rms of the whole field as well